app/services/journal_row_updater.rb

```diff
@@ -0,0 +1,36 @@
+class JournalRowUpdater
+
+  attr_accessor :order_detail, :journal_rows
+
+  def initialize(order_detail)
+    @order_detail = order_detail
+    @journal_rows = order_detail.journal_rows
+  end
+
+  def update
+    if recreate_journal_rows?
+      recreate_journal_rows
+    else
+      update_journal_rows
+    end
+    self
+  end
+
+  def update_journal_rows
+    journal_rows.each(&:update_amount)
+  end
+
+  def recreate_journal_rows
+    if journal_rows.present?
+      order_detail.journal_rows.destroy_all
+      JournalRowBuilder.new(order_detail.journal, [order_detail]).create
+    end
+  end
+
+  def recreate_journal_rows?
+    method = :recreate_journal_rows_on_order_detail_update
```

Comment:
If we define this method as returning `false` on the base `Account`, would that obviate the need for the `respond_to?` check?